##Objetivo: descrever o comportamento de cada variável do banco. A  meta é ter uma visão estatística completa do banco e entender sua estrutura antes de avançar para as perguntas do projeto. 

In [0]:
%sql
-- medidas descritivas de first_contact_date (amplitude das datas de primeiro contato) (Data)
SELECT
MIN(first_contact_date) AS primeiro_contato,
MAX(first_contact_date) AS ultimo_contato
FROM workspace.portfolio_1_olist.olist_marketing_qualified_leads_dataset;


primeiro_contato,ultimo_contato
2017-06-14,2018-05-31


In [0]:
-- medidas descritivas de first_contact_date (leads por dia da semana) (Data)
SELECT
date_format(first_contact_date, 'EEEE') AS dia_semana,
COUNT(*) AS leads
FROM workspace.portfolio_1_olist.olist_marketing_qualified_leads_dataset
GROUP BY 1
ORDER BY leads DESC;

dia_semana,leads
Monday,1495
Wednesday,1491
Tuesday,1452
Thursday,1354
Friday,1150
Sunday,531
Saturday,527


In [0]:
-- medidas descritivas de first_contact_date (leads por mês) (Data)
SELECT
date_trunc('month', first_contact_date) AS mes,
COUNT(*) AS leads
FROM workspace.portfolio_1_olist.olist_marketing_qualified_leads_dataset
GROUP BY 1
ORDER BY 1;

mes,leads
2017-06-01T00:00:00.000Z,4
2017-07-01T00:00:00.000Z,239
2017-08-01T00:00:00.000Z,386
2017-09-01T00:00:00.000Z,312
2017-10-01T00:00:00.000Z,416
2017-11-01T00:00:00.000Z,445
2017-12-01T00:00:00.000Z,200
2018-01-01T00:00:00.000Z,1141
2018-02-01T00:00:00.000Z,1028
2018-03-01T00:00:00.000Z,1174


Databricks visualization. Run in Databricks to view.

In [0]:
%sql
-- medidas descritivas de origin (Categórica)
SELECT
origin,
COUNT(*) AS qtd,
ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct,
ROUND(SUM(COUNT(*)) OVER (ORDER BY COUNT(*) DESC) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_acumulado
FROM workspace.portfolio_1_olist.olist_marketing_qualified_leads_dataset
GROUP BY origin
ORDER BY qtd DESC;

origin,qtd,pct,pct_acumulado
organic_search,2296,28.70,28.70
paid_search,1586,19.83,48.53
social,1350,16.88,65.40
unknown,1099,13.74,79.14
direct_traffic,499,6.24,85.38
email,493,6.16,91.54
referral,284,3.55,95.09
other,150,1.88,96.96
display,118,1.48,98.44
other_publicities,65,0.81,99.25


In [0]:
%sql
-- medidas descritivas de won_date (amplitude das datas de primeiro contato) (Data)
SELECT
MIN(won_date) AS primeiro_fechamento,
MAX(won_date) AS ultimo_fechamento
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset;

primeiro_fechamento,ultimo_fechamento
2017-12-05T02:00:00.000Z,2018-11-14T18:04:19.000Z


In [0]:
%sql
-- medidas descritivas de won_date (negocios fechados por dia da semana) (Data)
SELECT
date_format(won_date, 'EEEE') AS dia_semana,
COUNT(*) AS fechamentos
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY 1
ORDER BY fechamentos DESC;

dia_semana,fechamentos
Friday,192
Monday,184
Wednesday,167
Tuesday,159
Thursday,137
Saturday,3


In [0]:
%sql
-- medidas descritivas de won_date (negocios fechados por mes) (Data)
SELECT
date_trunc('month', won_date) AS mes,
COUNT(*) AS fechamentos
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY 1
ORDER BY 1;

mes,fechamentos
2017-12-01T00:00:00.000Z,3
2018-01-01T00:00:00.000Z,73
2018-02-01T00:00:00.000Z,113
2018-03-01T00:00:00.000Z,147
2018-04-01T00:00:00.000Z,207
2018-05-01T00:00:00.000Z,122
2018-06-01T00:00:00.000Z,57
2018-07-01T00:00:00.000Z,37
2018-08-01T00:00:00.000Z,33
2018-09-01T00:00:00.000Z,23


Databricks visualization. Run in Databricks to view.

In [0]:
%sql
-- medidas descritivas de business_segment (Categórica)
SELECT
business_segment,
COUNT(*) AS qtd,
ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct,
ROUND(SUM(COUNT(*)) OVER (ORDER BY COUNT(*) DESC) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_acumulado
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY business_segment
ORDER BY qtd DESC;

business_segment,qtd,pct,pct_acumulado
home_decor,105,12.47,12.47
health_beauty,93,11.05,23.52
car_accessories,77,9.14,32.66
household_utilities,71,8.43,41.09
construction_tools_house_garden,69,8.19,49.29
audio_video_electronics,64,7.60,56.89
computers,34,4.04,60.93
pet,30,3.56,64.49
food_supplement,28,3.33,67.81
food_drink,26,3.09,70.90


In [0]:
%sql
-- medidas descritivas de lead_type (Categórica)
SELECT
lead_type,
COUNT(*) AS qtd,
ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct,
ROUND(SUM(COUNT(*)) OVER (ORDER BY COUNT(*) DESC) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_acumulado
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY lead_type
ORDER BY qtd DESC;

lead_type,qtd,pct,pct_acumulado
online_medium,332,39.43,39.43
online_big,126,14.96,54.39
industry,123,14.61,69.00
offline,104,12.35,81.35
online_small,77,9.14,90.50
online_beginner,57,6.77,97.27
online_top,14,1.66,98.93
null,6,0.71,99.64
other,3,0.36,100.00


In [0]:
%sql
-- medidas descritivas de lead_behaviour_profile (Categórica)
SELECT
lead_behaviour_profile,
COUNT(*) AS qtd,
ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct,
ROUND(SUM(COUNT(*)) OVER (ORDER BY COUNT(*) DESC) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_acumulado
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY lead_behaviour_profile
ORDER BY qtd DESC;

lead_behaviour_profile,qtd,pct,pct_acumulado
cat,407,48.34,48.34
null,177,21.02,69.36
eagle,123,14.61,83.97
wolf,95,11.28,95.25
shark,24,2.85,98.10
"cat, wolf",8,0.95,99.05
"eagle, cat",3,0.36,99.76
"eagle, wolf",3,0.36,99.76
"shark, cat",1,0.12,100.00
"shark, wolf",1,0.12,100.00


In [0]:
%sql
-- medidas descritivas de has_company (Booleana)
SELECT
has_company,
COUNT(*) AS qtd,
ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct,
ROUND(SUM(COUNT(*)) OVER (ORDER BY COUNT(*) DESC) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_acumulado
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY has_company
ORDER BY qtd DESC;

has_company,qtd,pct,pct_acumulado
null,779,92.52,92.52
true,58,6.89,99.41
false,5,0.59,100.00


In [0]:
%sql
-- medidas descritivas de has_gtin (Booleana)
SELECT
has_gtin,
COUNT(*) AS qtd,
ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct,
ROUND(SUM(COUNT(*)) OVER (ORDER BY COUNT(*) DESC) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_acumulado
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY has_gtin
ORDER BY qtd DESC;

has_gtin,qtd,pct,pct_acumulado
null,778,92.40,92.40
true,54,6.41,98.81
false,10,1.19,100.00


In [0]:
%sql
-- medidas descritivas de average_stock (Categórica)
SELECT
average_stock,
COUNT(*) AS qtd,
ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct,
ROUND(SUM(COUNT(*)) OVER (ORDER BY COUNT(*) DESC) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_acumulado
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY average_stock
ORDER BY qtd DESC;

average_stock,qtd,pct,pct_acumulado
null,776,92.16,92.16
5-20,22,2.61,94.77
50-200,15,1.78,96.56
1-5,10,1.19,97.74
20-50,8,0.95,98.69
200+,7,0.83,99.52
unknown,4,0.48,100.00


In [0]:
%sql
-- medidas descritivas de business_type (Categórica)
SELECT
business_type,
COUNT(*) AS qtd,
ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct,
ROUND(SUM(COUNT(*)) OVER (ORDER BY COUNT(*) DESC) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_acumulado
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset
GROUP BY business_type
ORDER BY qtd DESC;

business_type,qtd,pct,pct_acumulado
reseller,587,69.71,69.71
manufacturer,242,28.74,98.46
null,10,1.19,99.64
other,3,0.36,100.00


In [0]:
%sql
-- medidas descritivas de declared_product_catalog_size (Numérica)
SELECT
COUNT(declared_product_catalog_size) AS valores_preenchidos,
ROUND(AVG(declared_product_catalog_size), 2) AS media,
ROUND(STDDEV(declared_product_catalog_size), 2) AS desvio_padrao,
MAX(declared_product_catalog_size) - MIN(declared_product_catalog_size) AS amplitude,
PERCENTILE(declared_product_catalog_size, 0.75) - PERCENTILE(declared_product_catalog_size, 0.25) AS iqr,
MIN(declared_product_catalog_size) AS minimo,
PERCENTILE(declared_product_catalog_size, 0.25) AS q1,
PERCENTILE(declared_product_catalog_size, 0.50) AS q2,
PERCENTILE(declared_product_catalog_size, 0.75) AS q3,
MAX(declared_product_catalog_size) AS maximo,
ROUND(SKEWNESS(declared_product_catalog_size), 2) AS assimetria,
ROUND(KURTOSIS(declared_product_catalog_size), 2) AS curtose
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset;

valores_preenchidos,media,desvio_padrao,amplitude,iqr,minimo,q1,q2,q3,maximo,assimetria,curtose
69,233.03,352.38,1999.0,270.0,1.0,30.0,100.0,300.0,2000.0,2.67,8.53


In [0]:
%sql
-- medidas descritivas de declared_monthly_revenue (Numérica)
SELECT
COUNT(declared_monthly_revenue) AS valores_preenchidos,
ROUND(AVG(declared_monthly_revenue), 2) AS media,
ROUND(STDDEV(declared_monthly_revenue), 2) AS desvio_padrao,
MAX(declared_monthly_revenue) - MIN(declared_monthly_revenue) AS amplitude,
PERCENTILE(declared_monthly_revenue, 0.75) - PERCENTILE(declared_monthly_revenue, 0.25) AS iqr,
MIN(declared_monthly_revenue) AS minimo,
PERCENTILE(declared_monthly_revenue, 0.25) AS q1,
PERCENTILE(declared_monthly_revenue, 0.50) AS q2,
PERCENTILE(declared_monthly_revenue, 0.75) AS q3,
MAX(declared_monthly_revenue) AS maximo,
ROUND(SKEWNESS(declared_monthly_revenue), 2) AS assimetria,
ROUND(KURTOSIS(declared_monthly_revenue), 2) AS curtose
FROM workspace.portfolio_1_olist.olist_closed_deals_dataset;

valores_preenchidos,media,desvio_padrao,amplitude,iqr,minimo,q1,q2,q3,maximo,assimetria,curtose
842,73377.68,1744799.18,5.0E7,0.0,0.0,0.0,0.0,0.0,5.0E7,27.99,795.63


## Observações: 

### variáveis numéricas

- A variável 'declared_monthly_revenue' possui preenchimento completo (842 registros), mas sua distribuição é extremamente concentrada em zero: primeiro quartil, mediana e terceiro quartil são iguais a 0. A média (73.377) é inflada por poucos valores extremos, chegando a um máximo de 50 milhões. A elevada assimetria (28) e curtose (796) confirmam forte influência de outliers, tornando a variável pouco informativa.

- A variável 'declared_product_catalog_size' apresenta uma distribuição plausível (mediana = 100; média = 233; assimetria = 2,67), porém conta com apenas 69 registros preenchidos entre os 842 negócios, comprometendo sua representatividade e utilidade analítica.

### variáveis temporais

- A captação de leads ('first_contact_date') ocorreu entre 14/06/2017 e 31/05/2018, enquanto os fechamentos ('won_date') foram registrados entre 05/12/2017 e 14/11/2018. Como os fechamentos continuaram por cerca de cinco meses após o encerramento da captação, os leads mais recentes tiveram menos tempo para converter do que os mais antigos. Essa diferença na janela de observação deve ser considerada nas análises posteriores. 

- A captação acelerou significativamente em 2018, com mais de mil leads por mês entre janeiro e maio, contra algumas centenas em 2017. Já os fechamentos cresceram até abril de 2018 (207 negócios) e passaram a cair nos meses seguintes. Essa redução reflete principalmente o menor tempo de maturação dos leads mais recentes, e não necessariamente uma piora no desempenho comercial.

- A análise por dia da semana mostra que a captação ocorre continuamente, inclusive aos fins de semana (cerca de 530 leads aos sábados e domingos). Em contrapartida, os fechamentos concentram-se em dias úteis, com apenas três registros aos sábados e nenhum aos domingos, indicando que a geração de leads é automatizada, enquanto a conversão depende da atuação da equipe comercial.

### variáveis categoricas

- A variável 'origin' (canal de captação) é liderada por 'organic_search' (28,7%), 'paid_search' (19,8%) e 'social' (16,9%). Contudo, 13,7% dos registros estão classificados como 'unknown' e 0,8% são nulos, totalizando cerca de 14,5% de leads sem origem identificada. Além disso, as categorias 'other' e 'other_publicities' (2,7% combinadas) possuem baixa especificidade. Esses valores devem ser tratados antes de comparações de conversão entre canais.

- A variável 'business_type' apresenta boa qualidade de dados. 'reseller' (69,7%) e 'manufacturer' (28,7%) concentram praticamente todos os registros, com apenas 1,2% de valores nulos e 0,4% classificados como 'other'. Trata-se da variável categórica mais consistente do banco.

- Na variável 'lead_type', predominam perfis online de diferentes portes ('online_medium', 'online_big', 'online_small', 'online_beginner' e 'online_top'), que juntos representam cerca de 72% dos registros. Os demais casos dividem-se entre 'industry' (14,6%) e 'offline' (12,3%).

- A variável 'business_segment' apresenta uma distribuição de cauda longa. Embora os segmentos mais frequentes, liderados por 'home_decor' e 'health_beauty', concentrem parcela relevante dos negócios, há um grande número de categorias com poucos registros, o que pode dificultar análises segmentadas.

- Por fim, 'lead_behaviour_profile' é um campo multivalorado, contendo combinações de perfis (por exemplo, 'cat, wolf'), além de apresentar 21% de valores ausentes. Como não está relacionado às perguntas analíticas deste projeto, o campo será mantido apenas para possíveis análises futuras.

### variáveis booleanas

- As variáveis 'has_company' e 'has_gtin' apresentam aproximadamente 92% de valores ausentes (779 e 778 registros, respectivamente), o que compromete sua utilidade analítica. Entre os poucos casos preenchidos, predominam valores verdadeiros. Esse padrão reforça a decisão de excluir ambas das análises.